### DISCUSSION TITLE
How to evaluate intensity of a line/vector across a pole figure?

### DISCUSSION THREAD

**Discussion by catcatfinishesaphd** [mtex-toolbox/mtex#1264] [Ask Anything] [answered]
Hi,

I would like to evaluate the intensity of a line/vector across a polefigure, using a pole figure as a variable.
I have looked in reference to: https://github.com/mtex-toolbox/mtex/discussions/806 & https://github.com/mtex-toolbox/mtex/discussions/696

However, both of these solutions use a ODF as the input to calculate the intensity. I would like to use a pole figure variable as an input. One of the solutions above indicates that I need to use calcDensity() but, I am not sure how to do this function to a pole figure.

`%% Plot the imported data
plot(pf)

%% Put the imported PF into "background" or "complete" pf
pf_complete = pf({2})
pf_background = pf({1})

% Subtract the background PF from the data PF's
pfback = correct(pf_complete, 'background', pf_background);
plot(pfback)
%% The colorcoding needs to be comparable between the plots (if multiple plots)
figure
plot(pfback, 'colorRange','equal')
%% saveas(gcf,'ColorComparePf.pdf')
v = vector3d.byPolar(linspace(0,pi/2,30),pi/2,'antipodal');
hold on
plot(v,'MarkerFaceAlpha',0.2)
hold off
nextAxis

%I know it will not evaluate the PF, not sure how to correct code..
plot(v.theta/degree,eval(pfback,v))
`

Thank you for any help on how to solve this issue!

**Comment by kilir** [comment, marked as answer]
Hi,
maybe there are various ways to do this:
Either compute a density estaimte:
```
mtexdata dubna
% just use the first one
pf = pf{1};

% compute a density estimate
pdf = calcDensity(pf.allR{:},'weights', pf.allI{:})

%evaluate somewhere
pdf.eval(vector3d([1 2 3]))
```

Alternatively, you can look for the point with a measurement in your pf closest to where you want to sample it or you can sample only points which exist in your pole figure grid.

Or you could estimate an odf in case you have a sufficient number of polefigures?

Hope that helps.

Cheers,
Rüdiger

**Comment by catcatfinishesaphd** [comment]
Hi Rüdiger,

This was very helpful, thank you! I only have a single pole figure for each material , so I can't estimate an ODF. 

I am now able to get the intensity of the values across the density estimate. However, I think that the when calculating the density estimate, some of the small differences between materials are lost. For example, when comparing pole figures directly (through mtex or the xrd software), I see a slight difference in intensity of the samples. But, when using calcDensity, this slight intensity difference disappears, and the calculated intensities are the same across all of my samples.

is there a way to increase the 'sensitivity' of calcdensity - or extract the values directly from the pole figure without using calcdensity?

**Reply by kilir** [reply]
How about looking for the closest measurement point?
```
mtexdata dubna
% pf = pf{1};% just use the first one

p = vector3d.byPolar(linspace(0,pi/2,30),-pi/3,'antipodal'); % points of interest



g = pf.allR{1}(:);
[~,id] = min(angle(p,g(:)))


d = pf.allI{1}(:);
val = d(id)


plot(g,d)
hold on
plot(p,'MarkerSize',3,'MarkerFaceColor','k')
hold off
nextAxis

plot(g(id),val,'MarkerSize',6,'MarkerFaceAlpha',0,'MarkerEdgeColor','k','Linewidth',2)

nextAxis
plot(p.theta/degree,val)
```
![blob](https://user-images.githubusercontent.com/11697557/139141951-6a3f9c2c-9ecf-4b65-8f28-f71543ad0222.png)

**Comment by catcatfinishesaphd** [comment]
Ahh yes! This works perfectly. Thank you very much!
